### DISCUSSION TITLE
Logging in FastAPI

### DISCUSSION THREAD

**Discussion by devankit01** [fastapi/fastapi#9726] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from main import celery
from models.glucose_models import GlucoseData
from config import session_main
from utils.logconf import log_config, LOGGER_NAME
import logging
log = logging.getLogger(LOGGER_NAME)

@celery.task
def send():
    session = session_main()
    query = session.query(GlucoseData).filter().limit(100)
    data = query.all()
    print(data)
    
    session.close()
    push_log("Async logging")
    return {"message" : True}

def push_log(message):
    log.info(message)
    return True
```

### Description

I am not able to do logging for celery task in FastAPI.

### Operating System

macOS

### FastAPI Version

fastapi==0.95.0 

### Python Version

3.9

**Comment by abhint** [comment]
It will help you. https://github.com/tiangolo/fastapi/discussions/7775#discussioncomment-5144123

**Reply by devankit01** [reply]
With the routers it's working but with the celery scheduled tasks is not working.

**Reply by abhint** [reply]
@devankit01 I think Celery tasks run in a separate process or thread, which can cause issues with logging configuration.
https://stackoverflow.com/questions/52086417/assistance-with-celery-logging

**Reply by devankit01** [reply]
Think so, but did not get any docs to do same thing and other thing the setthings are same as django but it do work in django.

**Reply by devankit01** [reply]
didn't worked it prints in console but not in the file.

**Reply by abhint** [reply]
try this, I think it will help. [structlog](https://github.com/hynek/structlog)

**Reply by devankit01** [reply]
okay..i'll try with this. Thanks
